src/wovnio/wovnphp/Headers.php

```diff
@@ -63,15 +63,20 @@ public function __construct(&$env, &$store)
         $this->unmaskedUrl = $this->protocol . '://' . $this->unmaskedHost . $this->unmaskedPathname;
         $this->host = $this->unmaskedHost;
         if ($store->settings['url_pattern_name'] === 'subdomain') {
-            $this->host = $this->removeLang($this->host, $this->lang());
+            $intermediateHost = explode('//', $this->removeLang($this->protocol . '://' . $this->host, $this->lang()));
+            $this->host = $intermediateHost[1];
         }
         if ($store->settings['use_proxy'] && isset($env['HTTP_X_FORWARDED_REQUEST_URI'])) {
             $clientRequestUri = $env['HTTP_X_FORWARDED_REQUEST_URI'];
         } else {
             $clientRequestUri = $env['REQUEST_URI'];
         }
         $exploded = explode('?', $clientRequestUri);
-        $this->pathname = $this->removeLang($exploded[0], $this->lang());
+        if ($store->settings['url_pattern_name'] === 'subdomain') {
+            $this->pathname = $exploded[0];
```

Comment:
Can't say I know the old code was doing here either. Is `$clientRequestUri` the path or full URL? Because I would expect `explode[0]` to include the host as well.

Reply:
It is always the path, like `/nice.html` or `/books/contemporary/fiction/`. The reason subdomain pattern is handled here differently is that we know for sure the path doesn't contain the lang when it's using subdomain pattern. The original logic screws up the path because the removeLang() can't handle a subdomain pattern without a full `scheme+host+path` combo.

Reply:
Wish the code made this more obvious 😅  When I see URI i'm going to assume it is a URI